.github/workflows/checks.yml

```diff
@@ -0,0 +1,89 @@
+name: Checks
+
+on:
+  - push:
+      branches:
+        main
```

Comment:
any reason not to run this on any branch push?

Reply:
It seems to me that only the `pull_request` checks are prominently visible in the UI and are likely to be noticed. I'd still like to have them run on pushes to main so that we can monitor if anything incorrect is merged, so that's why I added the push trigger. But otherwise in the usual case where you push a branch and open a PR almost immediately, we'd just be running a job that no-one will see.

That said, I'm not opposed to this; just think it's unnecessary.

Reply:
I don't think it's necessarily the usual case that you push a branch and _immediately_ open a PR, and either way, running an excess job is just extra cost for GitHub, not something that affects us.

Reply:
Fair enough. I can see this being useful for some workflows. (Maybe a manually triggered job would be ideal for that. Unfortunately, I looked into how to do those, but it's not entirely clear to me that it'd be anywhere near convenient in the UI.)

Reply:
workflow_dispatch is The Way but yeah, it's a bit arcane